static/app/views/issueList/draggableTabBar.tsx

```diff
@@ -24,17 +25,76 @@ export interface Tab {
 
 export interface DraggableTabBarProps {
   tabs: Tab[];
+  onAddView?: React.MouseEventHandler;
+  /**
+   * Callback function to be called when user clicks the `Delete` button.
+   * Note: The `Delete` button only appears for persistent views
+   */
   onDelete?: (key: MenuItemProps['key']) => void;
+  /**
+   * Callback function to be called when user clicks on the `Discard Changes` button.
+   * Note: The `Discard Changes` button only appears for persistent views when `isChanged=true`
+   */
   onDiscard?: (key: MenuItemProps['key']) => void;
+  /**
+   * Callback function to be called when user clicks on the `Discard` button for temporary views.
+   * Note: The `Discard` button only appears for temporary views
+   */
+  onDiscardTempView?: () => void;
+  /**
+   * Callback function to be called when user clicks the 'Duplicate' button.
+   * Note: The `Duplicate` button only appears for persistent views
+   */
   onDuplicate?: (key: MenuItemProps['key']) => void;
+  /**
+   * Callback function to be called when user clicks the 'Rename' button.
+   * Note: The `Rename` button only appears for persistent views
+   */
   onRename?: (key: MenuItemProps['key']) => void;
+  /**
+   * Callback function to be called when user clicks the 'Save' button.
+   * Note: The `Save` button only appears for persistent views when `isChanged=true`
+   */
   onSave?: (key: MenuItemProps['key']) => void;
+  /**
+   * Callback function to be called when user clicks the 'Save View' button for temporary views.
+   */
+  onSaveTempView?: () => void;
```

Comment:
Moving these to this component since there are three different possible variations of the tab menu options, and it makes more sense for this component to handle the logic that picks between them rather than directly inside the tabMenuButton component.